libs/features/mas/src/buildCheckoutUrl.js

```diff
@@ -0,0 +1,172 @@
+import { Landscape, WORKFLOW_STEP, PROVIDER_ENVIRONMENT } from './constants.js';
+
+const AF_DRAFT_LANDSCAPE = 'p_draft_landscape';
+const UCV3_PREFIX = '/store/';
+const PARAMETERS = new Map([
+  ['countrySpecific', 'cs'],
+  ['customerSegment', 'cs'],
+  ['quantity', 'q'],
+  ['authCode', 'code'],
+  ['checkoutPromoCode', 'apc'],
+  ['rurl', 'rUrl'],
+  ['curl', 'cUrl'],
+  ['ctxrturl', 'ctxRtUrl'],
+  ['country', 'co'],
+  ['language', 'lang'],
+  ['clientId', 'cli'],
+  ['context', 'ctx'],
+  ['productArrangementCode', 'pa'],
+  ['offerType', 'ot'],
+  ['marketSegment', 'ms'],
+]);
+/*?*
+ * List of the keys that can be used to construct Checkout URL.
+ * Allow-listing is required because some consumers of the library don't use Typescript so they can pass any object as
+ * 'checkoutData'. The library will ignore any unknown key in 'checkoutData'.
+ */
+const ALLOWED_KEYS = new Set([
```

Comment:
i know we copied it from pandora, but later we probably can remove this validation.
but for now lets keep as is